code/__HELPERS/bandetect.dm

```diff
@@ -0,0 +1,66 @@
+#define YOUNG 4
+
+
+/client/proc/join_date_check(var/jd)
+	if (!current_year)
+		current_year = text2num(time2text(world.realtime, "YYYY"))
+	if (!current_month)
+		current_month = text2num(time2text(world.realtime, "MM"))
+	if (!current_day)
+		current_day = text2num(time2text(world.realtime, "DD"))
+	var/warn = 0
+	var/y = text2num(copytext(jd, 1, 5))
+	var/m = text2num(copytext(jd, 6, 8))
+	var/d = text2num(copytext(jd, 9, 11))
+	if (current_month == 1 && current_day <= YOUNG)
+		if (y == current_year - 1 && m == 12 && d >= 31 - (YOUNG - current_day))
+			warn = 1
+		else if (y == current_year && m == 1)
+			warn = 1
+	else if (current_day <= YOUNG)
+		if (y == current_year)
+			if (m == current_month - 1 && d >= 28 - (YOUNG - current_day))
+				warn = 1
+			else if (m == current_month)
+				warn = 1
+	else if (y == current_year && m == current_month && d >= current_day - 4)
+		warn = 1
+	if (warn)
+		var/msg = "(IP: [address], ID: [computer_id]) has a recent join date of [jd]."
```

Comment:
this is a confusing message, we have a similar one for new players to our servers, this should explicitly say they're new to Byond.